### PR TITLE
Add local ec2

### PR DESCRIPTION
**Pull request by vkasljevic**

### CODE REVIEW

File: moto/vm_scripts/init_base.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+# Download base.vdi
+curl http://deps.memgraph.io/base.zip > base.zip
```
Comment (gitbuda) [inline moto/vm_scripts/init_base.sh]: Put the URL as a script parameter.

File: moto/ec2/models.py [hunk 2]
```diff
@@ -209,7 +212,8 @@ def __init__(self, ec2_backend, subnet, private_ip_address, device_index=0,
         self.instance = None
         self.attachment_id = None
 
-        self.public_ip = None
+        self.vm_id = vm_id
+        self._public_ip = None
```
Comment (gitbuda) [inline moto/ec2/models.py:218]: Why did you change the member name? It seems not a good idea.

**Review by gitbuda — CHANGES_REQUESTED** [review]

File: moto/ec2/models.py (hunk 2 above)
Comment (vkasljevic) [reply moto/ec2/models.py:218]: See property method public_id below.

File: moto/ec2/models.py [hunk 3]
```diff
@@ -407,6 +435,17 @@ def __init__(self, ec2_backend, image_id, user_data, security_groups, **kwargs):
         self.disable_api_termination = kwargs.get("disable_api_termination", False)
         self._spot_fleet_id = kwargs.get("spot_fleet_id", None)
         associate_public_ip = kwargs.get("associate_public_ip", False)
+
+        # Create VM instance
+        subprocess.run("vboxmanage clonevm --name instance_{} --register base"
```
Comment (gitbuda) [inline moto/ec2/models.py]: The user of Instance class has to be able to define does he want `vbox` or pure mocked instance.

**Review by gitbuda — CHANGES_REQUESTED** [review]

**Review by gitbuda — APPROVED** [review]
